MIGRATION.md

```diff
@@ -0,0 +1,86 @@
+0.19 to 0.21
+============
+
+## Configuration
+
+The biggest changes from 0.19 to 0.21 are related to the application.conf file, which has been restructured significantly.
+
+The configuration for backends now is all contained within a `backend` stanza, which specifies 1 stanza per name per backend and a default backend, as follows:
+
+```
+backend {
+    default=Local
+    providers {
+        Local {
+            actor-factory: "class path to BackendLifecycleActorFactory implementation"
+            config {
+                ... backend specific config ...
+            }
+        }
+        JES {
+            actor-factory: "class path to BackendLifecycleActorFactory implementation"
+            config {
+                ... backend specific config ...
+            }
+        }
+        SGE {
+            actor-factory: "class path to BackendLifecycleActorFactory implementation"
+            config {
+                ... backend specific config ...
+            }
+        }
+    }
+}
+```
+
+### Migrating a 0.19 config file to a 0.21 config file
+
+Copy `core/src/main/resources/application.conf` somewhere, and open up your old config file and copy the following values from the 0.19 config into the 0.21 config:
+
+Below is a table based diff, but there is also a [visual diff](http://i.imgur.com/i8j0Zvb.jpg) of the changes.
+
+|0.19|0.21|Comments|
+|----|----|--------|
+||`akka.actor.deployment`|Configures the number of threads to handle workflow log copying|
+||`akka.dispatchers.slow-actor-dispatcher`|Configures the "slow-actor" dispatcher|
+|`google.applicationName`|`google.application-name`||
+|`google.cromwellAuthenticationScheme`|removed|re-expressed as other configuration settings|
+|`google.serverAuth.pemFile`|`google.auths` with name `service-account`, then `pem-file`||
+|`google.serverAuth.serviceAccountId`|`google.auths` with name `service-account`, then `service-account-id`||
+|`google.userAuth`|removed||
+|`google.userAuthenticationScheme`|re-expressed as other configuration settings||
+|`google.refreshTokenAuth.client_id`|`google.auths` with name `user-via-refresh`, then `client-id`||
+|`google.refreshTokenAuth.client_secret`|`google.auths` with name `user-via-refresh`, then `client-secret`||
+||`engine.filesystems`|Defines additional filesystems that the engine is able to use to execute read and write WDL functions at the workflow-level|
+|`docker`|`backend.JES.config.dockerhub`|Moved to JES-specific configuration|
+|`jes.project`|`backend.JES.config.project`||
+|`jes.baseExecutionBucket`|`backend.JES.config.root`||
+|`jes.endpointUrl`|`backend.JES.config.genomics.endpoint-url`||
+|`jes.maximumPollingInterval`|`backend.JES.config.maximum-polling-interval`||
+|`shared-filesystem.root`|`backend.[Local|SGE].config.root`||
+|`shared-filesystem.localization`|`backend.[Local|SGE].config.filesystems.localization`||
+|`abortJobsInTerminate`|`system.abort-jobs-on-terminate`||
```

Comment:
This one also moved sections, it used to be in `backend` but now it's in `system`.